File: .archon/workflows/sdlc/verify-runtime/scripts/finish-attempt.py
"""Stop retries on cleanup failure and complete the final malformed attempt explicitly."""

import json
import os

result = json.loads(os.environ["INPUTS_ASSESSMENT"])
if os.environ["INPUTS_TEARDOWN_OK"] != "true":
    result.update(
        status="inconclusive",
        reason="teardown failed; external owner must clean up. "
        f"Prior assessment: {result['status']}: {result['reason']}",
    )
elif (
    result["status"] == "malformed"
    and int(os.environ["INPUTS_ATTEMPT"]) >= int(os.environ["INPUTS_ATTEMPT_LIMIT"])
):
    result.update(
        status="inconclusive",
        reason=f"malformed report after exhausting the retry budget: {result['reason']}",
    )
result.update(done=result["status"] != "malformed", checkout=os.environ["INPUTS_CHECKOUT"])
print(json.dumps(result))

File: .archon/workflows/sdlc/verify-runtime/scripts/finish-attempt.ts
const assessment: unknown = JSON.parse(process.env.INPUTS_ASSESSMENT ?? 'null');
if (assessment === null || typeof assessment !== 'object' || !('status' in assessment) ||
    !('reason' in assessment) || typeof assessment.status !== 'string' || typeof assessment.reason !== 'string') {
  throw new Error('runtime assessment is malformed');
}
let { status, reason } = assessment;
if (process.env.INPUTS_TEARDOWN_OK !== 'true') {
  reason = `teardown failed; external owner must clean up. Prior assessment: ${status}: ${reason}`;
  status = 'inconclusive';
} else if (status === 'malformed' && Number(process.env.INPUTS_ATTEMPT) >= Number(process.env.INPUTS_ATTEMPT_LIMIT)) {
  status = 'inconclusive';
  reason = `malformed report after exhausting the retry budget: ${reason}`;
}
console.log(JSON.stringify({ ...assessment, status, reason, done: status !== 'malformed', checkout: process.env.INPUTS_CHECKOUT }));
